File: result/metagpt/claude/humaneval/118.c
#include <stdio.h>
#include <string.h>

char* get_closest_vowel(const char* word) {
    static char out[2] = "";  // 用于返回结果的字符串
    const char* vowels = "AEIOUaeiou";
    int len = strlen(word);

    for (int i = len - 2; i >= 1; i--) {
        if (strchr(vowels, word[i]) != NULL) {
            if (strchr(vowels, word[i + 1]) == NULL && strchr(vowels, word[i - 1]) == NULL) {
                out[0] = word[i];
                out[1] = '\0';
                return out;
            }
        }
    }

    return out;  // 如果没有找到符合条件的元音，返回空字符串
}

File: result/metagpt/gpt4o/humaneval/118.rs
fn get_closest_vowel(word: &str) -> String {
    let vowels = "AEIOUaeiou";
    let len = word.len();
    let chars: Vec<char> = word.chars().collect();

    for i in (1..len - 1).rev() {
        if vowels.contains(chars[i]) {
            if !vowels.contains(chars[i + 1]) && !vowels.contains(chars[i - 1]) {
                return chars[i].to_string();
            }
        }
    }

    String::new() // Return an empty string if no suitable vowel is found
}

fn main() {
    let word = "example";
    let closest_vowel = get_closest_vowel(word);
    println!("Closest vowel: {}", closest_vowel);
}